CMakeLists.txt

```diff
@@ -78,6 +78,9 @@ else(MSVC)
   check_cxx_compiler_flag("-std=c++11"    SUPPORT_CXX11)
   set(CMAKE_C_FLAGS "-O2 -Wall -fPIC ${CMAKE_C_FLAGS}")
   set(CMAKE_CXX_FLAGS "-O2 -Wall -fPIC -std=c++11 ${CMAKE_CXX_FLAGS}")
+  if (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 7.0)
```

Comment:
Need to confirm if compiler is g++, otherwise the rule is not correct fo clang

Reply:
@tqchen I have used 'g++' compiler (cc1plus) and warnings are seen.

I verified build with 'clang' and not seen the these warnings.
Shall I add check for 'g++' here?

Reply:
yes, please only trigger this flag when compiler is g++